advancedblacklist/commands/blacklist.py

```diff
@@ -0,0 +1,171 @@
+        """Edit the reason for a user in the blacklist.
+
+        **Arguments**
+            - `user` The user to edit the reason of.
+        """
+        if user.bot:
+            return await ctx.send("That user is a bot.")
+        elif await self.bot.is_owner(user):
+            return await ctx.send("That user is a bot owner.")
+        elif not await in_blacklist(self.bot, user.id, ctx.guild):
+            return await ctx.send("That user is not in the blacklist.")
+        try:
+            await edit_reason(self.bot, user, reason, False)
+        except KeyError:
+            return await ctx.send("That user was not in the blacklist.")
+        await ctx.send("Done. Edited the reason for that user.")
+
+    @blacklist.command(name="list")
+    async def blacklist_list(self, ctx: commands.Context):
+        """List the users in the blacklist."""
+        if not (bl := await get_blacklist(self.bot)):
+            return await ctx.send("There are no users in the blacklist.")
+        msg = "Blacklisted Users:"
+        for key, value in bl.items():
+            name = u.name if (u := self.bot.get_user(key)) else "Unknown User"
+            msg += f"\n\t- [{key}] {name}: {value}"
+        await ctx.send_interactive(pagify(msg, page_length=1800), "yml")
+
+    @commands.group(name="localblacklist", aliases=["localblocklist"])
+    @commands.admin_or_permissions(manage_guild=True)
+    @commands.guild_only()
+    async def local_blacklist(self, ctx: commands.Context):
+        """Manage the local blacklist for your guild."""
+        pass
+
+    @local_blacklist.command(name="add")
+    async def local_blacklist_add(
+        self, ctx: commands.Context, users: commands.Greedy[discord.Member], *, reason: str = None
+    ):
+        """Add users to the local blacklist
+
+        **Arguments**
+            - `users` The users to add to the local blacklist. These cannot be bots
+            - `reason` The reason for adding these users to the blacklist. This is optional
+        """
+        if not users:
+            raise commands.UserInputError
+        for user in users:
+            if user.bot:
+                return await ctx.send("You cannot add bots to the local blacklist.")
+            elif await self.bot.is_owner(user):
+                return await ctx.send("You cannot add a bot owner to the local blacklist.")
+            elif user.id == ctx.guild.owner_id:
+                return await ctx.send("You cannot add the guild owner to the local blacklist.")
+            elif ctx.author == user:
+                return await ctx.send("You cannot add yourself to the local blacklist.")
+        reason = reason or "No reason provided."
+        await add_to_blacklist(self.bot, users, reason, guild=ctx.guild)
+        that = "that user" if len(users) == 1 else "those users"
+        await ctx.send(f"Done. Added {that} to the local blacklist.")
```

Comment:
```suggestion
        if not members:
            raise commands.UserInputError
        for member in members:
            if member.bot:
                return await ctx.send("You cannot add bots to the local blacklist.")
            elif await self.bot.is_owner(member ):
                return await ctx.send("You cannot add a bot owner to the local blacklist.")
            elif member.id == ctx.guild.owner_id:
                return await ctx.send("You cannot add the guild owner to the local blacklist.")
            elif ctx.author == member:
                return await ctx.send("You cannot add yourself to the local blacklist.")
        reason = reason or "No reason provided."
        await add_to_blacklist(self.bot, members, reason, guild=ctx.guild)
        that = "that member" if len(users) == 1 else "those members"
        await ctx.send(f"Done. Added {that} to the local blacklist.")
```
Worth seeing my review comment above about this. Also updated variable names to keep consistency within the code. Annoyingly, this github review won't let me do it all in one go. I'll leave a review comment for each command (1/4)